src/trix/models/composition.coffee

```diff
@@ -78,16 +78,22 @@ class Trix.Composition extends Trix.BasicObject
     block = document.getBlockAtIndex(index)
 
     if block.getBlockBreakPosition() is offset
-      document = document.removeTextAtRange(range)
-      range = [position, position]
+      if block.text.getStringAtRange([offset - 1, offset]) is "\n"
```

Comment:
Consider extracting this into a local variable so the expression reads more like prose:
```coffee
previousCharacter = block.text.getStringAtRange([offset - 1, offset])
if previousCharacter is "\n"
  # ...
```